pkg/utils/maputils.go

```diff
@@ -84,3 +86,37 @@ func SetDifference[K comparable, T1, T2 any](newSet map[K]T2, oldSet map[K]T1) (
 
 	return added, removed
 }
+
+// ConvertToSetFromMap converts a map to a set. The set is represented as a map with the key as the element of the set
+func ConvertToSetFromMap[K comparable, V any](set map[K]struct{}, source map[K]V) {
+	for k := range source {
+		set[k] = struct{}{}
+	}
+}
+
+// ConvertToSetFromSlice converts a slice to a set. The set is represented as a map with the key as the element of the set
+func ConvertToSetFromSlice[K comparable](set map[K]struct{}, source []K) {
+	for _, k := range source {
+		set[k] = struct{}{}
+	}
+}
+
+func AddToSet[K comparable](set map[K]struct{}, key K) {
+	if _, ok := set[key]; !ok {
+		set[key] = struct{}{}
+	}
+}
```

Comment:
nit: you don't need this check `if _, ok := set[key]; !ok {` because there's only one possible value in the map for a given key. So this function would be just `set[key] = struct{}{}`, so I don't really see the need for this function (at least, not unless it was a method on a new `type Set struct {...}`)